### ISSUE TITLE
Playback on other apps (eg Spotify) not stopped when playing song

### ISSUE THREAD

**Issue by jakobkukla** [dkanada/gelli#62]
If you listen to another media source (for example Spotify) and click on a song from an album, the playback from the other app doesn't always stop. This only happens if I click on a song directly. Not if I click the Play/Pause button directly.

**Comment by dkanada** [comment]
@jakobkukla is this still an issue? It looks similar to #90 which I consider resolved.

**Comment by jakobkukla** [comment]
Just tried out 1.3.1 and can confirm this is still happening yes.

**Comment by jakobkukla** [comment]
Sooo I found the problem. It's because Gelli doesn't request audio focus if you click on a song directly. It invokes openQueue with startPlaying=true but not the normal play() function in MusicService, which is the only place where we invoke requestFocus().

While trying to fix it, I found out that ExoPlayer can manage audio focus for us. I'm working on a PR which would remove a lot of the audio focus handling code. Would such a PR be accepted? :)
